File: src/arrays_and_strings/java/CheckPermutation.java
package arrays_and_strings.java;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

/**
 *
 */
public class CheckPermutation {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    String a = in.next();
    String b = in.next();

    System.out.println(isPermutation(a, b) ? "Yes" : "No");
  }

  private static boolean isPermutation(String a, String b) {
    int length = a.length();
    if (length != b.length()) return false;
    Map<Character, Integer> buf = new HashMap<>();
    for (char c : a.toCharArray()) {
      buf.computeIfPresent(c, (k, v) -> v + 1);
      buf.putIfAbsent(c, 1);
    }
    for (char c : b.toCharArray()) {
      buf.computeIfPresent(c, (k, v) -> v - 1);
      buf.putIfAbsent(c, -1);
    }
    for (Map.Entry<Character, Integer> entry : buf.entrySet()) {
      if (entry.getValue() != 0) return false;
    }

    return true;
  }
}

File: src/arrays_and_strings/kotlin/CheckPermutation.kt
package arrays_and_strings.kotlin

import java.util.Scanner

/**
 *
 */
fun main(args: Array<String>) {
  val sc = Scanner(System.`in`)
  val a = sc.next()
  val b = sc.next()
  print(if (isPermutation(a, b)) "Yes" else "No")
}

fun isPermutation(a: String, b: String): Boolean {
  val length = a.length
  if (length != b.length) return false
  val buf = HashMap<Char, Int>()
  a.forEach {
    buf.computeIfPresent(it, {_, v -> v + 1})
    buf.putIfAbsent(it, 1)
  }
  b.forEach {
    buf.computeIfPresent(it, {_, v -> v - 1})
    buf.putIfAbsent(it, -1)
  }
  for ((_, v) in buf) {
    if (v != 0) return false
  }

  return true
}